client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -583,36 +610,49 @@ private static void addSuppressedException(Exception suppressedException, Except
-                return addRequestBody(new HttpDeleteWithEntity(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpDeleteWithEntity(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpGetWithEntity.METHOD_NAME:
-                return addRequestBody(new HttpGetWithEntity(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpGetWithEntity(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpHead.METHOD_NAME:
-                return addRequestBody(new HttpHead(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpHead(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpOptions.METHOD_NAME:
-                return addRequestBody(new HttpOptions(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpOptions(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpPatch.METHOD_NAME:
-                return addRequestBody(new HttpPatch(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpPatch(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpPost.METHOD_NAME:
                 HttpPost httpPost = new HttpPost(uri);
-                addRequestBody(httpPost, entity, compressionEnabled);
+                addRequestBody(httpPost, entity, compressionEnabled, chunkedTransferEncodingEnabled);
                 return httpPost;
             case HttpPut.METHOD_NAME:
-                return addRequestBody(new HttpPut(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpPut(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             case HttpTrace.METHOD_NAME:
-                return addRequestBody(new HttpTrace(uri), entity, compressionEnabled);
+                return addRequestBody(new HttpTrace(uri), entity, compressionEnabled, chunkedTransferEncodingEnabled);
             default:
                 throw new UnsupportedOperationException("http method not supported: " + method);
         }
     }
 
-    private static HttpRequestBase addRequestBody(HttpRequestBase httpRequest, HttpEntity entity, boolean compressionEnabled) {
+    private static HttpRequestBase addRequestBody(
+        HttpRequestBase httpRequest,
+        HttpEntity entity,
+        boolean compressionEnabled,
+        boolean chunkedTransferEncodingEnabled
+    ) {
         if (entity != null) {
             if (httpRequest instanceof HttpEntityEnclosingRequestBase) {
                 if (compressionEnabled) {
-                    entity = new ContentCompressingEntity(entity);
+                    entity = new ContentCompressingEntity(entity, chunkedTransferEncodingEnabled);
+                } else {
+                    entity = new ContentHttpEntity(entity, chunkedTransferEncodingEnabled);
```

Comment:
We probably could do `if (entity.isChunked() != chunkedTransferEncodingEnabled) { .. }` and do wrapping, otherwise we should be good to go as-is , right?

Reply:
@reta by default entity.isChunked returns false

Reply:
> @reta by default entity.isChunked returns false

@jiten1551 sorry, not sure I get it, `HttpEntity` is an interface, we deal with some of its implementation, where is the default you are referring to?

Reply:
see https://github.com/opensearch-project/OpenSearch/blob/main/client/rest-high-level/src/main/java/org/opensearch/client/RequestConverters.java#L755

here NByteArrayEntity is created for all request which extends AbstractHttpEntity which has default fault value of is chunked false as we're not setting it anywhere
https://www.javadoc.io/static/org.apache.httpcomponents/httpcore-nio/4.4.10/org/apache/http/nio/entity/NByteArrayEntity.html

Reply:
We should not rely on any implementation details here, we deal with interface. Anyway, the question is: why we need to wrap if the `entity` into `ContentHttpEntity` if chunking settings are not changing?

Reply:
@reta  we introduced a flag which says enable/disable chunkedEncoding so we're wrapping it in under some entity to support that originally for compression we're using chunked and for non-compressed we're using non-chunked

Reply:
@jiten1551 I understand what this flag means:
 - if `entity.isChunked() == false` and `chunkedTransferEncodingEnabled == false`, why we need to wrap?
 - if `entity.isChunked() == true` and `chunkedTransferEncodingEnabled == true`, why we need to wrap?

Reply:
> in those cases we don't need to wrap

that's what I am suggesting to check here, nothing else (whatever assumptions may be deducted from implementation today may change in the future).

Reply:
I changed the code to use an `Optional` and fallback to the `entity.isChunked`. It's a bit messy, lmk what you think?

Reply:
I like it this way better (they dealing with `null`s fe)